kraken-java-sdk/kraken-java-sdk-sync/src/main/java/com/consoleconnect/kraken/operator/sync/service/push/PushAPIActivityLogScheduler.java

```diff
@@ -0,0 +1,194 @@
+package com.consoleconnect.kraken.operator.sync.service.push;
+
+import static com.consoleconnect.kraken.operator.core.toolkit.JsonToolkit.fromJson;
+import static com.consoleconnect.kraken.operator.core.toolkit.PagingHelper.toPageNoSubList;
+import static java.util.Collections.emptyList;
+
+import com.consoleconnect.kraken.operator.core.dto.ApiActivityLog;
+import com.consoleconnect.kraken.operator.core.dto.ComposedHttpRequest;
+import com.consoleconnect.kraken.operator.core.entity.AbstractHttpEntity;
+import com.consoleconnect.kraken.operator.core.entity.ApiActivityLogEntity;
+import com.consoleconnect.kraken.operator.core.entity.MgmtEventEntity;
+import com.consoleconnect.kraken.operator.core.enums.EventStatusType;
+import com.consoleconnect.kraken.operator.core.enums.MgmtEventType;
+import com.consoleconnect.kraken.operator.core.exception.KrakenException;
+import com.consoleconnect.kraken.operator.core.mapper.ApiActivityLogMapper;
+import com.consoleconnect.kraken.operator.core.model.HttpResponse;
+import com.consoleconnect.kraken.operator.core.repo.ApiActivityLogRepository;
+import com.consoleconnect.kraken.operator.core.repo.MgmtEventRepository;
+import com.consoleconnect.kraken.operator.sync.model.SyncProperty;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+@ConditionalOnProperty(
+    value = "app.control-plane.push-activity-log-external.enabled",
+    havingValue = "true")
+public class PushAPIActivityLogScheduler {
```

Comment:
This class can  extend KrakenServerConnector and reuse the common http request handler. 
For example,  can refer to the class com.consoleconnect.kraken.operator.sync.service.PushHeartbeatService.

Reply:
@kuangxiang20240501 
the change applied and pushed